include/mbedtls/ssl.h

```diff
@@ -2220,6 +2226,27 @@ void mbedtls_ssl_set_timer_cb( mbedtls_ssl_context *ssl,
                                mbedtls_ssl_set_timer_t *f_set_timer,
                                mbedtls_ssl_get_timer_t *f_get_timer );
 
+#if defined(MBEDTLS_SSL_SRV_C)
+#if defined(MBEDTLS_SSL_SERVER_NAME_INDICATION)
```

Comment:
:rocket: As this is a new callback, existing practice would have had to register certificates in `mbedtls_ssl_config` if `MBEDTLS_SSL_SERVER_NAME_INDICATION` support was not present.  Still, I agree with your thought "in principle one could imagine the callback being useful in other contexts".  I'll remove the `MBEDTLS_SSL_SERVER_NAME_INDICATION` check since the server certificate callback may find use by application developers as a callback which necessarily occurs after all TLS extensions in Client Hello have been processed, and can be used to set the server certificate (if not already set in `mbedtls_ssl_config`).  This might also be a good place to check for TLS extensions required by an application, but which are missing from the Client Hello, detectable by the application if the application has used callbacks for when those TLS extensions exist.